### ISSUE TITLE
Maximum number of clients reached src/clientdll/inputgenerator_linux.cpp (336) : m_pDisplay

### ISSUE THREAD

**Issue by Nocty1501** [ValveSoftware/steam-for-linux#9435]
I've been having the same issue as in #8595, running Linux Mint 21.1 Vera, Cinnamon (though I replaced the DE with KDE Plasma). My system is up-to-date.

Steam client version is 1682723851 (in the open beta currently).

Steam loads, updates, and runs fine for a few minutes (10-20) before quietly crashing, generating a crash file in `/tmp/dumps`. Whatever game was running at the time freezes/lags for a few seconds before also crashing.

No Steam client beta, installing/uninstalling `libnm0:i386` didn't make a difference. I'm using an NVIDIA GTX 970 with `nvidia-driver-525`.

Note: `nvidia-driver-530` and entering the Steam client beta did not change anything, still crashes at regular intervals.

I caught a crash running from terminal, here are the last few lines:
```
Maximum number of clients reachedsrc/clientdll/inputgenerator_linux.cpp (336) : m_pDisplay
src/clientdll/inputgenerator_linux.cpp (336) : m_pDisplay
assert_20230501062906_35.dmp[23697]: Uploading dump (out-of-process)
/tmp/dumps/assert_20230501062906_35.dmp
crash_20230501062906_36.dmp[23700]: Uploading dump (out-of-process)
/tmp/dumps/crash_20230501062906_36.dmp
assert_20230501062906_35.dmp[23697]: Finished uploading minidump (out-of-process): success = yes
assert_20230501062906_35.dmp[23697]: response: CrashID=bp-7e42426d-375e-42cf-8c32-b71ca2230430
assert_20230501062906_35.dmp[23697]: file ''/tmp/dumps/assert_20230501062906_35.dmp'', upload yes: ''CrashID=bp-7e42426d-375e-42cf-8c32-b71ca2230430''
crash_20230501062906_36.dmp[23700]: Finished uploading minidump (out-of-process): success = yes
crash_20230501062906_36.dmp[23700]: response: CrashID=bp-e910535c-a2ef-4a35-9c6d-c7ee62230430
crash_20230501062906_36.dmp[23700]: file ''/tmp/dumps/crash_20230501062906_36.dmp'', upload yes: ''CrashID=bp-e910535c-a2ef-4a35-9c6d-c7ee62230430''
/home/nocty1501/.local/share/Steam/steam.sh: line 798: 14965 Segmentation fault      (core dumped) "$STEAMROOT/$STEAMEXEPATH" "$@"
```

**Comment by Nocty1501** [comment]
Here are the crash logs in my `/tmp/dumps` folder.
[dumps.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/11366629/dumps.tar.gz)

**Comment by kisak-valve** [comment]
Hello @Nocty1501, for reference, crash_20230501062906_36.dmp is a SIGSEGV in libX11.so.6.4.0.

**Comment by Nocty1501** [comment]
I'm afraid I don't know what I could do with that information :(

Following the steps taken in #8595, I ran the following:

```
$ sudo apt policy libx11-6 libx11-6:i386
libx11-6:
  Installed: 2:1.7.5-1
  Candidate: 2:1.7.5-1
  Version table:
 *** 2:1.7.5-1 500
        500 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages
        100 /var/lib/dpkg/status
libx11-6:i386:
  Installed: 2:1.7.5-1
  Candidate: 2:1.7.5-1
  Version table:
 *** 2:1.7.5-1 500
        500 http://archive.ubuntu.com/ubuntu jammy/main i386 Packages
        100 /var/lib/dpkg/status
```

I attempted to uninstall `libx11-6:i386`, but Steam just requested to reinstall it next time I ran it. I couldn't uninstall `libx11-6`.

**Comment by Nocty1501** [comment]
I should also mention that for the entire time the Steam client runs, it spams the terminal with the following:

```
!! Steam controller device opened for index 0.
Local Device Found
  type: 045e 028e
  path: sdl://273
  serial_number:  - 0
  Manufacturer: 
  Product:      X360 Controller
  Release:      110
  Interface:    -1

!! Steam controller device opened for index 0.
Controller device closed after hid_read failure
Local Device Found
  type: 2563 0537
  path: sdl://274
  serial_number:  - 0
  Manufacturer: 
  Product:      Trust-GMP-04
  Release:      111
  Interface:    -1

!! Steam controller device opened for index 0.
Controller device closed after hid_read failure
Local Device Found
  type: 045e 028e
  path: sdl://275
  serial_number:  - 0
  Manufacturer: 
  Product:      X360 Controller
  Release:      110
  Interface:    -1

!! Steam controller device opened for index 0.
Controller device closed after hid_read failure
Local Device Found
  type: 045e 028e
  path: sdl://276
  serial_number:  - 0
  Manufacturer: 
  Product:      X360 Controller
  Release:      110
  Interface:    -1

!! Steam controller device opened for index 0.
Controller device closed after hid_read failure
Local Device Found
  type: 045e 028e
  path: sdl://277
  serial_number:  - 0
  Manufacturer: 
  Product:      X360 Controller
  Release:      110
  Interface:    -1
```

I have the USB component of a wireless gamepad plugged in, but the gamepad itself is turned off and I am not using it.

**Comment by TTimo** [comment]
Hello,

> I have the USB component of a wireless gamepad plugged in, but the gamepad itself is turned off and I am not using it.

If you unplug the various controllers you may have completely, does the problem go away? Can you narrow it down to a specific one?

**Comment by Nocty1501** [comment]
> 

I was thinking this exact same thing and unplugged the gamepad. It's been a few hours and Steam hasn't crashed! I'm going to leave this issue open for now in case it does come back (though I'm not going to plug the gamepad back in), just in case it's a coincidence.

**Comment by TTimo** [comment]
It's a legitimate issue, I don't know that you should really close it because you have a workaround. Just trying to narrow down what is happening here.

**Comment by TTimo** [comment]
Could you pull a list of open file descriptors while the problem is happening? We'd like to match the spam you're seeing in the terminal with X11 connections left open.

You'd lookup steam's pid with `pgrep -x steam` then grab the output of `ls -1al /proc/<pid>/fd`

If you don't want to post publicly feel free to email to ttimo@valvesoftware.com - thanks!

**Comment by Nocty1501** [comment]
> Could you pull a list of open file descriptors while the problem is happening? We'd like to match the spam you're seeing in the terminal with X11 connections left open.
> 
> You'd lookup steam's pid with `pgrep -x steam` then grab the output of `ls -1al /proc/<pid>/fd`
> 
> If you don't want to post publicly feel free to email to [ttimo@valvesoftware.com](mailto:ttimo@valvesoftware.com) - thanks!

Sent!

**Comment by TTimo** [comment]
Fix for this went in with last night's beta. Please retest.

**Comment by Nocty1501** [comment]
So it seems the crashes have ceased, but the console spam hasn't, though it has changed in content. 

The following is the "common" error message that shows up about every 2 seconds:

```
Local Device Found
  type: 045e 028e
  path: sdl://619
  serial_number:  - 0
  Manufacturer: 
  Product:      X360 Controller
  Release:      110
  Interface:    -1

!! Steam controller device opened for index 0.
Controller device closed after hid_read failure
!! Controller 0 attributes:
  Type: 31
  ProductID: 654
  Serial: 45e-28e-f18050f
  Capabilities: 001843ff
  Firmware Version: 0
  Firmware Build Time: 2147483647 (Tue, 19 Jan 2038 03:14:07 GMT)
  Bootloader Build Time: 2147483647 (Tue, 19 Jan 2038 03:14:07 GMT)
Loaded Config for Local Selection Path for App ID 413090, Controller 0: controller_base/bigpicture_gamepad.vdf
Loaded Config for Local Selection Path for App ID 413090, Controller 0: controller_base/bigpicture_gamepad.vdf
Loaded Config for Local Selection Path for App ID 769, Controller 0: controller_base/basicui_gamepad.vdf
Loaded Config for Last Resort Path for App ID 769, Controller 0: /home/nocty1501/.local/share/Steam//controller_base/basicui_gamepad.vdf
Loaded Config for Local Override Path for App ID 413080, Controller 0: /home/nocty1501/.local/share/Steam//controller_base/empty.vdf
Controller 0 disconnected
```

Then, occasionally, once every 20-30 error messages, this one will appear instead:

```
Local Device Found
  type: 2563 0537
  path: sdl://620
  serial_number:  - 0
  Manufacturer: 
  Product:      Trust-GMP-04
  Release:      111
  Interface:    -1

!! Steam controller device opened for index 0.
Controller device closed after hid_read failure
!! Controller 0 attributes:
  Type: 30
  ProductID: 1335
  Serial: 2563-537-f18050f
  Capabilities: 001803f7
  Firmware Version: 0
  Firmware Build Time: 2147483647 (Tue, 19 Jan 2038 03:14:07 GMT)
  Bootloader Build Time: 2147483647 (Tue, 19 Jan 2038 03:14:07 GMT)
Loaded Config for Local Selection Path for App ID 413090, Controller 0: controller_base/bigpicture_gamepad.vdf
Loaded Config for Local Selection Path for App ID 413090, Controller 0: controller_base/bigpicture_gamepad.vdf
Loaded Config for Local Selection Path for App ID 769, Controller 0: controller_base/basicui_gamepad.vdf
Loaded Config for Last Resort Path for App ID 769, Controller 0: /home/nocty1501/.local/share/Steam//controller_base/basicui_gamepad.vdf
Loaded Config for Local Override Path for App ID 413080, Controller 0: /home/nocty1501/.local/share/Steam//controller_base/empty.vdf
Controller 0 disconnected
```

The error messages appear to always be identical in content, with the exception of the `path:` field, which increments with every message (regardless of error "type"). I still only have one controller plugged in.

**Comment by TTimo** [comment]
Thanks. We think your driver is incorrectly advertising a controller that it can't read from (that's expected, since it's powered off). We fixed Steam so it doesn't crash to resource exhaustion in such situations. The driver would need to be fixed, but that's not us :)

**Comment by Nocty1501** [comment]
The controller wasn't for any Steam games to begin with, I just wanted to leave it plugged in :p

Cheers, guys!

**Comment by Nocty1501** [comment]
I should mention that the constant re-reading of what controller is plugged in is causing significant stuttering which ends up feeling like a really bad lag spike every few seconds. This is so bad it happens even when no game is open, which for a while lead me to think it was my driver or DE acting up. I had to unplug the controller to make it stop because it made certain shooters unplayable due to the constant stuttering affecting my ability to react.

I'm leaving this comment because this was never an issue on Windows 7 (which I used prior to moving to Linux Mint), so presumably this is not intended behavior.
